Initial implementation (#1)
appuio/component-openshift4-splunk-forwarder: docs/modules/ROOT/pages/references/parameters.adoc

The parent key for all of the following parameters is `openshift4_splunk_forwarder`.

== `fluentd`

[horizontal]
type:: dictionary
default::
+
[source,yaml]
----
fluentd:
    loglevel: info
    replicas: 2
    sharedkey: splunkforwarding
    ssl:
      enabled: false
      cert: ""
      key: ""
      passphrase: ""
----

Fluentd pod parameters.

=== `loglevel`

[horizontal]
type:: string
default:: "info"

Log level of the splunk-forwarder pod (https://docs.fluentd.org/deployment/logging).

=== `replicas`

[horizontal]
type:: integer
default:: 2

Number of replicas of the splunk-forwarder pod.

=== `sharedkey`

[horizontal]
type:: string
default:: "splunkforwarding"

The shared key for authentication (see https://docs.fluentd.org/input/forward).

=== `ssl.enabled`

[horizontal]
type:: boolean
default:: false

Enable fluentd ssl encryption.

=== `ssl.cert`

[horizontal]
type:: string
default:: ""

> String of ssl certificate.

Reviewer: ```suggestion
The SSL certificate as a string.
```

=== `ssl.key`

[horizontal]
type:: string
default:: ""

String of ssl private key.

=== `ssl.passphrase`

[horizontal]
type:: string
default:: ""

Passphrase of ssl private key.

== `splunk`

[horizontal]
type:: dictionary
default::
+
[source,yaml]
----
splunk:
  token: 4a8a737d-5452-426c-a6f7-106dca4e813f
  hostname: splunk.syn-splunk.svc
  port: 8088
  insecure: true
  protocol: https
  index: openshift
  sourcetype: openshift
  source: openshift
  ca: ""
----

Parameters for splunk integration.

=== `token`

[horizontal]
type:: string
default:: "4a8a737d-5452-426c-a6f7-106dca4e813f"

> Splunk HEC token.

Reviewer: You should document how to get such a token (either here or in the getting started guide)

      passphrase: '?{vaultkv:${cluster:tenant}/${cluster:name}/openshift4-splunk-forwarder/${_instance}/fluentd/privatekey-passphrase}'
----

Fluentd pod parameters.

=== `loglevel`

[horizontal]
type:: string
default:: "info"

Log level of the splunk-forwarder pod (https://docs.fluentd.org/deployment/logging).

=== `replicas`

[horizontal]
type:: integer
default:: 2

Number of replicas of the splunk-forwarder pod.

=== `sharedkey`

[horizontal]
type:: string
default:: Vault reference

A Vault reference pointing to the Vault secret containing the shared key for authentication (see https://docs.fluentd.org/input/forward).

[source,bash]
----
# Adjust to your environment
instance=#INSTANCE_NAME
key="clusters/kv/${TENANT_ID}/${CLUSTER_ID}/openshift4-splunk-forwarder/${instance}/fluentd"

# Query for existing secrets first
vault kv get "${key}"

# If there are existing secrets, add your instance secrets:
vault kv patch "${key}" sharedkey=$(pwgen -s 32 1)

# If there is no pre-existing secret, create new:
vault kv put "${key}" sharedkey=$(pwgen -s 32 1)
----

=== `ssl.enabled`

[horizontal]
type:: boolean
default:: false

> Enable fluentd ssl encryption.

Reviewer: ```suggestion
Enable fluentd SSL encryption.
```

=== `ssl.cert`

[horizontal]
type:: string
default:: ""

The SSL certificate as a string.

See xref:how-tos/create-fluentd-cert.adoc[Create Fluentd Cert] to generate the Fluentd certificate.

=== `ssl.key`

[horizontal]
type:: string
default:: Vault reference

> The SSL private key as a string.

Reviewer: ```suggestion
A Vault reference to the SSL private key.
```

See xref:how-tos/create-fluentd-cert.adoc[Create Fluentd Cert] to generate the Fluentd certificate.

== `splunk`

[horizontal]
type:: dictionary
default::
+
[source,yaml]
----
splunk:
  token: #VAULT_SECRET
  hostname: splunk.syn-splunk.svc
  port: 8088
  insecure: true
  protocol: https
  index: openshift
  sourcetype: openshift
  source: openshift
  ca: ""
----

> Parameters for splunk integration.

Reviewer: ```suggestion
Parameters for Splunk integration.
```

From the official website, it looks like Splunk should be capitalized

=== `token`

[horizontal]
type:: string
default:: Vault reference

A Vault reference pointing to the Vault secret containing the Splunk HEC token.

See xref:how-tos/get-splunk-token.adoc[Get Splunk Token] to acquire the HEC token.

=== `hostname`

[horizontal]
type:: string
default:: "splunk.syn-splunk.svc"

Hostname of Splunk server.

=== `port`

[horizontal]
type:: integer
default:: 8088

Port of Splunk server.

=== `insecure`

[horizontal]
type:: boolean
default:: true

> If `insecure: true` certificates of Splunk server aren't validated.

Reviewer: ```suggestion
If this value is set to `true` the certificates of Splunk server aren't validated.
```

=== `protocol`

[horizontal]
type:: string
default:: "https"

Protocol used to forward logs to Splunk server, either `http` or `https`.

=== `index`

[horizontal]
type:: string
default:: "openshift"

See https://docs.splunk.com/Splexicon:Index.

=== `sourcetype`

[horizontal]
type:: string
default:: "openshift"

See https://docs.splunk.com/Splexicon:Sourcetype.

=== `source`

[horizontal]
type:: string
default:: "openshift"

See https://docs.splunk.com/Splexicon:Source.

=== `ca`

[horizontal]
type:: string
default:: ""

> Splunk servers CA certificate.

Reviewer: ```suggestion
The CA certificate for the Splunk server.
Only required if the Splunk server is secured with a certificate which is signed by a CA that's not trusted by default.
```
